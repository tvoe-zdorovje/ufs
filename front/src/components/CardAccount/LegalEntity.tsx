import React from 'react'
import {
  Row,
  Col
} from 'antd'
import renderPair from './RenderPair'
import { LegalEntity } from 'interfaces/dto/Account.interface'

export interface LegalEntityProps {
  legalEntity: LegalEntity
}

export default function LegalEntity({ legalEntity }: LegalEntityProps) {
  const pairs = [['Наименование', legalEntity.fullName],
    ['Сокращенное наименование', legalEntity.shortName],
    ['ИНН', legalEntity.inn],
    ['ОГРН', legalEntity.ogrn],
    ['КПП', legalEntity.kpp],
    ['Юридический адрес', legalEntity.legalAddress],
    ['Фактический адрес', legalEntity.factAddress]]

  return (
    <Row>
      <Col span={23}>
        {pairs.map((item, i) => renderPair(item[0], item[1], i))}
      </Col>
    </Row>
  )
}
