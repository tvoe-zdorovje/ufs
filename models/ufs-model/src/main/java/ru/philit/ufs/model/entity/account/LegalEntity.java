package ru.philit.ufs.model.entity.account;

import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;
import ru.philit.ufs.model.entity.common.ExternalEntity;

/**
 * Сущность Юридического лица.
 */
@EqualsAndHashCode(of = {"id"}, callSuper = false)
@ToString
@Getter
@Setter
public class LegalEntity extends ExternalEntity {

  private String id;
  private String shortName;
  private String fullName;
  private String inn;
  private String ogrn;
  private String kpp;
  private String legalAddress;
  private String factAddress;

}
