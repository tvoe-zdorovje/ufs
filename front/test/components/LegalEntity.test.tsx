import React from 'react'
import { render } from 'enzyme'
import toJson from 'enzyme-to-json'

import LegalEntity, { LegalEntityProps } from 'components/CardAccount/LegalEntity'

describe('Компонент LegalEntity', () => {

  const props: LegalEntityProps = {
    legalEntity: {
      id: '', fullName: 'OOO ДНК', shortName: 'OOO ДНК', inn: '01221412', ogrn: '102312412', kpp: '121412412',
      legalAddress: 'Ярославль', factAddress: 'Ярославль'
    }
  }

  const renderWrapper = render(<LegalEntity {...props} />)

  it('совпадает со своим снимком', () => {
    expect(toJson(renderWrapper)).toMatchSnapshot()
  })
})
