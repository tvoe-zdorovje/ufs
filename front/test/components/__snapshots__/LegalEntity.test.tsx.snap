// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент LegalEntity совпадает со своим снимком 1`] = `
<div
  class="ant-row"
>
  <div
    class="ant-col-23"
  >
    <div
      class="ant-row pairRow"
    >
      <div
        class="ant-col-8 rightAligned bold"
      >
        Наименование
      </div>
      <div
        class="ant-col-16"
      >
        OOO ДНК
      </div>
    </div>
    <div
      class="ant-row pairRow"
    >
      <div
        class="ant-col-8 rightAligned bold"
      >
        Сокращенное наименование
      </div>
      <div
        class="ant-col-16"
      >
        OOO ДНК
      </div>
    </div>
    <div
      class="ant-row pairRow"
    >
      <div
        class="ant-col-8 rightAligned bold"
      >
        ИНН
      </div>
      <div
        class="ant-col-16"
      >
        01221412
      </div>
    </div>
    <div
      class="ant-row pairRow"
    >
      <div
        class="ant-col-8 rightAligned bold"
      >
        ОГРН
      </div>
      <div
        class="ant-col-16"
      >
        102312412
      </div>
    </div>
    <div
      class="ant-row pairRow"
    >
      <div
        class="ant-col-8 rightAligned bold"
      >
        КПП
      </div>
      <div
        class="ant-col-16"
      >
        121412412
      </div>
    </div>
    <div
      class="ant-row pairRow"
    >
      <div
        class="ant-col-8 rightAligned bold"
      >
        Юридический адрес
      </div>
      <div
        class="ant-col-16"
      >
        Ярославль
      </div>
    </div>
    <div
      class="ant-row pairRow"
    >
      <div
        class="ant-col-8 rightAligned bold"
      >
        Фактический адрес
      </div>
      <div
        class="ant-col-16"
      >
        Ярославль
      </div>
    </div>
  </div>
</div>
`;
